test/nn/test_load_state_dict.py

```diff
@@ -362,7 +365,162 @@ def test_load_state_dict_warn_assign(self):
             m.load_state_dict(state_dict)
 
 
+class MyLoadTensor(torch.Tensor):
+
+    # enable deepcopy
+    def new_empty(self, shape):
+        return type(self)(shape)
+
+    @classmethod
+    def __torch_function__(cls, func, types, args=(), kwargs=None):
+        kwargs = {} if kwargs is None else kwargs
+
+        def module_load_from(dest, src):
+            # always convert src to MyLoadTensor
+            if type(src) is torch.Tensor:
+                return MyLoadTensor(src)
+            elif type(src) is MyLoadTensor:
+                return src
+            elif type(src) is MyWrapperLoadTensor:
+                return MyLoadTensor(src._data)
+            else:
+                raise NotImplementedError(f"module_load_from(): unsupported type(src) {type(src)}")
+
+        def module_load_to(src, dest):
+            if type(dest) in {torch.nn.Parameter, torch.Tensor, MyLoadTensor, MyWrapperLoadTensor}:
+                return src
+            else:
+                raise NotImplementedError(f"module_load_to(): unsupported type(dest) {type(dest)}")
+
+        if func is torch.Tensor.module_load_from:
+            return module_load_from(*args, **kwargs)
+        elif func is torch.Tensor.module_load_to:
+            return module_load_to(*args, **kwargs)
+        elif func is torch.Tensor.detach:
+            return MyLoadTensor(args[0].data.detach())
+        else:
+            with torch._C.DisableTorchFunctionSubclass():
+                return func(*args, **kwargs)
+
+
+class MyWrapperLoadTensor(torch.Tensor):
+
+    @staticmethod
+    def __new__(cls, data: torch.Tensor):
+        t = torch.Tensor._make_wrapper_subclass(
+            cls, data.size(),
+            dtype=data.dtype, layout=data.layout,
+            device=data.device, requires_grad=data.requires_grad,
+            strides=data.stride(), storage_offset=data.storage_offset())
+        return t
+
+    def __init__(self, data: torch.Tensor):
+        self._data = data
+
+    def __repr__(self):
+        return f"MyWrapperLoadTensor({self._data.__repr__()})"
+
+    # enable deepcopy
+    def new_empty(self, shape):
```

Comment:
Similar: why?

Reply:
I think this was wrong, `new_empty` should only be needed for non-wrapper subclasses, my bad